src/frontegg-auth/src/unvalidated_auth.rs

```diff
@@ -0,0 +1,155 @@
+#![warn(clippy::todo)]
+#![warn(clippy::wildcard_dependencies)]
+#![warn(clippy::zero_prefixed_literal)]
+#![warn(clippy::borrowed_box)]
+#![warn(clippy::deref_addrof)]
+#![warn(clippy::double_must_use)]
+#![warn(clippy::double_parens)]
+#![warn(clippy::extra_unused_lifetimes)]
+#![warn(clippy::needless_borrow)]
+#![warn(clippy::needless_question_mark)]
+#![warn(clippy::needless_return)]
+#![warn(clippy::redundant_pattern)]
+#![warn(clippy::redundant_slicing)]
+#![warn(clippy::redundant_static_lifetimes)]
+#![warn(clippy::single_component_path_imports)]
+#![warn(clippy::unnecessary_cast)]
+#![warn(clippy::useless_asref)]
+#![warn(clippy::useless_conversion)]
+#![warn(clippy::builtin_type_shadow)]
+#![warn(clippy::duplicate_underscore_argument)]
+#![warn(clippy::double_neg)]
+#![warn(clippy::unnecessary_mut_passed)]
+#![warn(clippy::wildcard_in_or_patterns)]
+#![warn(clippy::collapsible_if)]
+#![warn(clippy::collapsible_else_if)]
+#![warn(clippy::crosspointer_transmute)]
+#![warn(clippy::excessive_precision)]
+#![warn(clippy::overflow_check_conditional)]
+#![warn(clippy::as_conversions)]
+#![warn(clippy::match_overlapping_arm)]
+#![warn(clippy::zero_divided_by_zero)]
+#![warn(clippy::must_use_unit)]
+#![warn(clippy::suspicious_assignment_formatting)]
+#![warn(clippy::suspicious_else_formatting)]
+#![warn(clippy::suspicious_unary_op_formatting)]
+#![warn(clippy::mut_mutex_lock)]
+#![warn(clippy::print_literal)]
+#![warn(clippy::same_item_push)]
+#![warn(clippy::useless_format)]
+#![warn(clippy::write_literal)]
+#![warn(clippy::redundant_closure)]
+#![warn(clippy::redundant_closure_call)]
+#![warn(clippy::unnecessary_lazy_evaluations)]
+#![warn(clippy::partialeq_ne_impl)]
+#![warn(clippy::redundant_field_names)]
+#![warn(clippy::transmutes_expressible_as_ptr_casts)]
+#![warn(clippy::unused_async)]
+#![warn(clippy::disallowed_methods)]
+#![warn(clippy::disallowed_macros)]
+#![warn(clippy::disallowed_types)]
+#![warn(clippy::from_over_into)]
+// END LINT CONFIG
+
+use std::ops::Sub;
+
+use tokio::sync::Mutex;
+
+use crate::{ApiTokenResponse, AppPassword, Client, Error, REFRESH_SUFFIX};
+use chrono::{DateTime, Duration, Utc};
+
+pub struct UnvalidatedAuthenticationConfig {
+    pub admin_api_token_url: String,
+    pub app_password: AppPassword,
+}
+
+pub struct UnvalidatedAuthentication {
+    pub admin_api_token_url: String,
+    pub client: Client,
+    pub auth: Mutex<Option<ApiTokenResponse>>,
+    pub app_password: AppPassword,
+}
+
+impl UnvalidatedAuthentication {
+    /// Creates a new frontegg authentication without signature validation
+    /// and handles the access token for a unique `app_password`.
+    /// `UnvalidatedAuthentication` should be NEVER used on the server side.
+    /// It should only be used in the client side. E.g.: Front-end apps, CLI, etc.
+    /// Unvalidated authentication avoids carrying certs to validate the tokens.
+    pub fn new(config: UnvalidatedAuthenticationConfig, client: Client) -> Self {
```

Comment:
Because we should only use this for client side apps, should we move this into a separate `client` module, or something, to make it super obvious that this shouldn't be used server side? We could also mark it as `unsafe` to let callers know that they need to assert some invariants, i.e. this is only used client side. What do you think?